substrate/frame/staking-async/src/session_rotation.rs

```diff
@@ -751,6 +751,13 @@ impl<T: Config> Rotator<T> {
 	fn cleanup_old_era(starting_era: EraIndex) {
 		EraElectionPlanner::<T>::cleanup();
 
+		let diff = T::BondingDuration::get();
+		if starting_era >= diff {
+			let target_era = starting_era - diff;
+			TotalUnbondInEra::<T>::remove(target_era);
+			EraLowestRatioTotalStake::<T>::remove(target_era);
```

Comment:
`prune_era` might be a better place for this. This function only cleans up data that we maintain per era cycle, and not historical.